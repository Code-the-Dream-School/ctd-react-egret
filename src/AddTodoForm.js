import React from "react";
import {v4 as uuidv4} from "uuid";

const AddTodoForm = ({onAddTodo}) => {
  const [todoTitle, setTodoTitle] = React.useState("");
  const handleTitleChange = (e) => {
    const newTodoTitle = e.target.value;
    setTodoTitle(newTodoTitle);
  };
  const handleAddTodo = (e) => {
    onAddTodo({id: uuidv4(), title: todoTitle});
    setTodoTitle("");
    e.preventDefault();
  };
  return (
    <form onSubmit={handleAddTodo}>
      <label htmlFor='todoTitle'>Title: </label>
      <input
        id='todoTitle'
        name='title'
        type='text'
        value={todoTitle}
        onChange={handleTitleChange}
      />
      <button type='submit'>Add</button>
    </form>
  );
};

export default AddTodoForm;
