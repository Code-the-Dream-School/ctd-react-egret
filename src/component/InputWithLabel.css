#todoTitle {
 display: flex;
 justify-content: center;
 margin:auto;
 align-items: center;
 width: 80%;
}
#todoTitle:focus{
 outline: none;
 border-color: cadetblue;

}