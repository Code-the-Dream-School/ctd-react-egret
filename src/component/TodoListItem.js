import React, { useState } from 'react'
import styles from './TodoListItem.module.css';
import { AiFillDelete } from "react-icons/ai";
import PropTypes from 'prop-types';
function TodoListItem({ todo, onRemoveTodo }) {
  const [checked, setChecked] = useState({ labelChecked: false });
  const labelRef = React.createRef();
  const handleClick = e => {
    if (checked.labelChecked === false) {
      labelRef.current.style.textDecoration = "line-through";
    } else {
      labelRef.current.style.textDecoration = "none";
    }
    setChecked({ labelChecked: !checked.labelChecked });
  };
  TodoListItem.propTypes = {
    todo: PropTypes.object,
    onRemoveTodo: PropTypes.func
  }
  return (
    <div >
      <li className={styles.ListItem}>
        <label ref={labelRef} htmlFor="strikethrough">
          <input type="checkbox" id="strikethrough" name="strikethrough" onClick={handleClick} className={styles.checkbox1} />
          &nbsp;&nbsp;
          {todo.fields.Title}
        </label>
        <button className={styles.removeButton} type="button" onClick={() => onRemoveTodo(todo.id)}>
          <AiFillDelete size={20} style={{ fill: 'brown' }} />
        </button>
      </li>
    </div>
  )
}


export default TodoListItem
