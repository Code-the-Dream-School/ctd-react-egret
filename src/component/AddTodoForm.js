import React from 'react';
import InputWithLabel from './InputWithLabel';
import Button from '@mui/material/Button';
import styled from './AddTodoForm.module.css'
import PropTypes from 'prop-types';

function AddTodoForm({ onAddTodo }) {
    const [todoTitle, setTodoTitle] = React.useState("");
    const handleTitleChange = (event) => {
        const newTodoTitle = event.target.value;
        setTodoTitle(newTodoTitle);
    };
    const handleAddTodo = (event) => {
        event.preventDefault();
        if (todoTitle === "") {
            return false;
        }
        setTodoTitle("");
    };
    AddTodoForm.propTypes = {
        onAddTodo: PropTypes.func
      }
    return (
        <form onSubmit={handleAddTodo} className = {styled.form}>
            <InputWithLabel todoTitle = {todoTitle} handleTitleChange= {handleTitleChange}>Title</InputWithLabel>
            <Button  type="submit">Add</Button>
        </form>
    );
}

export default AddTodoForm;
