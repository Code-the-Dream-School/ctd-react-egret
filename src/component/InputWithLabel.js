
import React, { useEffect, useRef } from "react";
import  './InputWithLabel.css';
import PropTypes from 'prop-types';
const InputWithLabel = ({ todoTitle, handleTitleChange, children }) => {
  const inputRef = useRef();
  
  useEffect(() => {
    inputRef.current.focus();
  });
  InputWithLabel.propTypes = {
    children: PropTypes.string,
    todoTitle: PropTypes.string,
    handleTitleChange: PropTypes.func
  }

  return (
    <>
      <label htmlFor="todoTitle" >{children}</label>
      <input 
      
        ref={inputRef}
        id="todoTitle"
        type="text"
        placeholder = "Enter Task"
        name="title"
        value={todoTitle}
        onChange={handleTitleChange}
      />
    </>
  );
};

export default InputWithLabel;