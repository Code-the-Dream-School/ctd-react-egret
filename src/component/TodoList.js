import React from "react";
import TodoListItem from "./TodoListItem";
import PropTypes from 'prop-types';
function TodoList({ todoList, onRemoveTodo }) {
  
  TodoList.propTypes = {
    todoList: PropTypes.array,
    onRemoveTodo: PropTypes.func
  }
  return (
    <ul>
      {todoList.map((todo) => (
        <TodoListItem key={todo.id} todo={todo} onRemoveTodo={onRemoveTodo} />
      ))}
    </ul>
  );
}

export default TodoList;
