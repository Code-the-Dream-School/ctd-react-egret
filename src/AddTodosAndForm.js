import AddTodoForm from "./AddTodoForm";
import TodoList from "./TodoList";
import styles from "./AddTodosAndForm.module.css";

const AddTodosAndForm = ({
  onAddTodo,
  onRemoveTodo,
  todoList,
  fetchStatus,
  children,
}) => (
  <div className={styles.todostWrapper}>
    <div className={styles.homeButton}>{children[0]}</div>
    <div className={styles.listWrapper}>
      <h2>{children[1]}</h2>
      {fetchStatus.isError && (
        <p>
          <strong>SOMETHING WENT WRONG:</strong>&nbsp;
          {fetchStatus.errMsg}
          {/* <strong>SOMETHING WENT WRONG:</strong>&nbsp;
          {fetchStatus.errMsg.error}--{fetchStatus.errMsg.message} */}
        </p>
      )}
      {fetchStatus.isLoading ? (
        <p>Loading ...</p>
      ) : (
        <TodoList
          todoList={todoList}
          listName={children[1]}
          onRemoveTodo={onRemoveTodo}
        />
      )}
      <AddTodoForm listName={children[1]} onAddTodo={onAddTodo} />
    </div>
  </div>
);

export default AddTodosAndForm;
