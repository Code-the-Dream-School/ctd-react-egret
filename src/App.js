import React from 'react';

import TodoList from "./TodoList";
import AddTodoform from "./AddTodoForm";
function App() {
 
  const[todoList, setTodoList] = React.useState([]);
const addTodo = (newTodo) => {
  setTodoList ([...todoList, newTodo])
};
function removeTodo(id) {
  const newTodoList = todoList.filter(function(item){
    return item.id !==id
  })
  setTodoList(newTodoList)
};
 
  return (
    <div>
      <h1>Todo List</h1> 
      <AddTodoform onAddTodo={addTodo}/>
  
      <TodoList onRemoveTodo ={removeTodo} todoList = {todoList}/>
     
    </div>
  );
}

export default App;
