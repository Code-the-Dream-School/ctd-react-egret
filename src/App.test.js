import * as React from "react";
import {
  render,
  screen,
  fireEvent,
  userEvent,
  act,
} from "@testing-libraray/react";
import App, {todoListReducer} from "./App";
import AddTodosAndForm from "./AddTodosAndForm";
import AddTodoForm from "./AddTodoForm";
import InputWithLabel from "./InputWithLabel";
import TodoList from "./TodoList";
import TodoListItem from "./TodoListItem";

/*
Declare Mock Test Data
*/

const readingTodos = [
  {
    id: 1,
    fields: {
      Task: "Pull all the changes from origin to local main",
    },
  },
  {
    id: 2,
    fields: {
      Task: "Create a new branch lesson-4-1 to do work",
    },
  },
];

const homeworkTodos = [
  {
    id: 3,
    fields: {
      Task: "Read all the Road to React chapters assigned for this week",
    },
  },
  {
    id: 4,
    fields: {
      Task: "Complete all the exercises prescribed in the Road to React book",
    },
  },
];

/*
Test Suites
*/

// describe("App", () => { });

// For todoListReducer in App component

describe("todoListReducer", () => {
  test("fetch new readingTodos pending", () => {
    const action = {type: "TODOLIST_FETCH_INIT"};
    const state = {
      data: [],
      isLoading: true,
      isError: false,
    };
    const newState = todoListReducer(state, action);
    const expectedState = {
      data: [],
      isLoading: true,
      isError: false,
    };

    expect(newState).toStrictEqual(expectedState);
  });
});
