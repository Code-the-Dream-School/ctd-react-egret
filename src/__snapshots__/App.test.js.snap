// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddTodoForm renders snapshot 1`] = `undefined`;

exports[`FireRing renders snapshot 1`] = `undefined`;

exports[`InputWithLabel renders snapshot 1`] = `undefined`;

exports[`SideBar renders snapshot 1`] = `undefined`;

exports[`TodoCard renders snapshot 1`] = `undefined`;

exports[`TodoContainer renders snapshot 1`] = `undefined`;

exports[`TodoList renders snapshot 1`] = `undefined`;

exports[`TodoListItem renders snapshot 1`] = `undefined`;
